test/common/upstream/outlier_detection_impl_test.cc

```diff
@@ -394,23 +397,27 @@ TEST(OutlierDetectionEventLoggerImplTest, All) {
   std::shared_ptr<MockHostDescription> host(new NiceMock<MockHostDescription>());
   ON_CALL(*host, cluster()).WillByDefault(ReturnRef(cluster));
   NiceMock<MockSystemTimeSource> time_source;
+  Optional<SystemTime> time;
 
   EXPECT_CALL(log_manager, createAccessLog("foo")).WillOnce(Return(file));
   EventLoggerImpl event_logger(log_manager, "foo", time_source);
 
   std::string log1;
-  EXPECT_CALL(*file, write("{\"time\": \"1970-01-01T00:00:00.000Z\", \"cluster\": "
+  EXPECT_CALL(host->outlier_detector_, lastUnejectionTime()).WillOnce(ReturnRef(time));
+  EXPECT_CALL(*file, write("{\"time\": \"1970-01-01T00:00:00.000Z\", \"secs_since_last_action\": "
+                           "\"-1\", \"cluster\": "
```

Comment:
you need to have a test which actually returns a time. You aren't testing that path.